.github/workflows/pull-request-account-compression.yml

```diff
@@ -0,0 +1,94 @@
+name: Account Compression Pull Request
+
+on:
+  pull_request:
+    paths:
+      - "account-compression/**"
+      - "libraries/concurrent-merkle-tree/**"
+      - "ci/*-version.sh"
+      - "ci/anchor-cliversion.sh"
+      - ".github/workflows/pull-request-account-compression.yml"
+  push:
+    branches: [master]
+    paths:
+      - "account-compression/**"
+      - "libraries/concurrent-merkle-tree/**"
+      - "ci/*-version.sh"
+      - "ci/anchor-cliversion.sh"
+      - ".github/workflows/pull-request-account-compression.yml"
+
+jobs:
+  anchor-build-account-compression:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Set env vars
+        run: |
+          source ci/rust-version.sh
+          echo "RUST_STABLE=$rust_stable" >> $GITHUB_ENV
+          source ci/solana-version.sh
+          echo "SOLANA_VERSION=$solana_version" >> $GITHUB_ENV
+          source ci/anchor-cliversion.sh
+          echo "ANCHOR_CLI_VERSION=$anchor_cli_version" >> $GITHUB_ENV
+
+      - uses: actions-rs/toolchain@v1
+        with:
+          toolchain: ${{ env.RUST_STABLE }}
+          override: true
+          profile: minimal
+
+      - uses: actions/cache@v2
+        with:
+          path: |
+            ~/.cargo/registry
+            ~/.cargo/git
+          key: cargo-build-${{ hashFiles('**/Cargo.lock') }}-${{ env.RUST_STABLE }}
+
+      - uses: actions/cache@v2
+        with:
+          path: |
+            ~/.cargo/bin/rustfilt
+          key: cargo-sbf-bins-${{ runner.os }}
+
+      - uses: actions/cache@v2
+        with:
+          path: ~/.cache/solana
+          key: solana-${{ env.SOLANA_VERSION }}
+
+      - name: Install dependencies
+        run: |
+          ./ci/install-build-deps.sh
+          ./ci/program-anchor-build.sh
```

Comment:
for clarity, let's break this out into a separate step, since it's building the program